apis/src/responses/tournament.rs

```diff
@@ -0,0 +1,104 @@
+use crate::responses::user::UserResponse;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct TournamentAbstractResponse {
+    pub id: Uuid,
+    pub nanoid: String,
+    pub name: String,
+}
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub struct TournamentResponse {
```

Comment:
Could be worth breaking this up into more structs for readability and also usability in the future

Reply:
Do you have a suggestion how to break this up?

Reply:
Worth noting that `TimeDetails` and `BandDetails` are common to the ChallangeDetails type so maybe we can find a way to integrate these. Also this is a suggestion only feel free to use some other pattern

```
pub struct Participats{
    pub invitees: Vec<UserResponse>,
    pub players: Vec<UserResponse>,
    pub organizers: Vec<UserResponse>,
}
pub struct TournamentDate {
    pub start: Option<DateTime<Utc>>,
    pub created: DateTime<Utc>,
    pub updated: DateTime<Utc>,
}

pub struct TimeDetails {
    pub mode: String,
    pub base: Option<i32>,
    pub increment: Option<i32>,
}

pub struct BandDetails {
    pub upper: Option<i32>,
    pub lower: Option<i32>,
}
pub struct TournamentResponse {
    pub id: Uuid,
    pub nanoid: String,
    pub name: String,
    pub description: String,
    pub participants: Participats,
    pub date: TournamentDate,
    pub time: TimeDetails,
    pub band: BandDetails,
    pub scoring: String,
    pub tiebreaker: Vec<String>,
    pub seats: i32,
    pub rounds: i32,
    pub joinable: bool,
    pub invite_only: bool,
    pub mode: String,
}
```